client/src/components/CommentsContainer/index.tsx

```diff
@@ -1,27 +1,159 @@
+        UserId: 1,
+        EventId: 1,
+        User: {
+          id: 1,
+          username: '',
+          profileImg: '',
+        },
+      },
+    ],
+  }
+
+  const initOneCommentValue: IOneComment = {
+    message: '',
+    data:
+      {
+        id: 0,
+        content: '',
+        image: '',
+        createdAt: '',
+        updatedAt: '',
+        UserId: 1,
+        EventId: 1,
+        User: {
+          id: 1,
+          username: '',
+          profileImg: '',
+        },
+      },
+
+  }
+
+  const [comments, setComments] = useState<IComments>(initCommentsValue)
+  const [newComment, setNewComments] = useState<IOneComment>(initOneCommentValue)
+  const [showMore, setShowMore] = useState<number>(1)
+  const [allComments, setAllComments] = useState<IComments[]>([])
+  const [hasMore, setHasMore] = useState<boolean>(true)
+  const [open, setOpen] = useState<boolean>(false)
+  const [error, setError] = useState<boolean | string>(false)
+  const [loader, setLoader] = useState<boolean>(true)
+
+  const handleOpen = ():void => setOpen(true)
+  const handleClose = ():void => setOpen(false)
+  const handleShowMore = ():void => setShowMore(showMore + 1)
+
+  const idParams = useParams().id
+
+  React.useEffect(() => {
+    (async (): Promise<void> => {
+      try {
+        const result = await ApiService.get(`/events/${idParams}/comments?offset=${showMore}`)
+        setComments(result.data)
+        setAllComments([...allComments, ...result.data.data])
+        if (!result.data.data.length) setHasMore(false)
+        setLoader(false)
+      } catch (err) {
+        setError('We\'re having some errors in getting the information. We\'re working on it.')
+        setLoader(false)
+      }
+    })()
+  }, [idParams, showMore])
+
+  React.useEffect(() => {
+    setComments({ ...comments, data: [newComment.data, ...comments.data] })
+  }, [newComment])
+
+  if (loader) {
+    return (
+      <CircularProgress sx={{ margin: '200px auto', display: 'block' }} />
+    )
+  }
+
+  if (error) {
+    return (
+      <Alert severity="error" sx={{ width: '80%', margin: '20px auto' }}>
+        <AlertTitle>Error</AlertTitle>
+        {error}
+      </Alert>
+    )
+  }
+
+  return (
+    <div className="comments-container">
+      <Button
+        onClick={handleOpen}
+        variant="contained"
+        sx={{ backgroundColor: '#2A2A2A' }}
+      >
+        {/* <AddOutlinedIcon /> */}
+        Add Comments
+      </Button>
+
+      { !comments.data.length
+      && (
+      <Alert severity="info" sx={{ width: '80%', margin: '50px auto' }}>
+        <AlertTitle>no comments</AlertTitle>
+        there is no comments yet for this event
+      </Alert>
+      )}
+      <InfiniteScroll
+        dataLength={allComments.length}
+        next={handleShowMore}
+        hasMore={hasMore}
+        loader={<CircularProgress sx={{ margin: '10px auto', display: 'block' }} />}
+        endMessage={<p className="end-message">There are no more Comments</p>}
```

Comment:
here you are viewing 2 messages if there is no comments.
stick by one of them, may be with the `InfiniteScroll` one